src/base/dense/owl_base_dense_matrix_generic.ml

```diff
@@ -0,0 +1,19 @@
+(*
+ * OWL - OCaml Scientific and Engineering Computing
+ * Copyright (c) 2016-2019 Liang Wang <liang.wang@cl.cam.ac.uk>
+ *)
+
+include Owl_base_dense_ndarray_generic
```

Comment:
Logically I think perhaps we should also move matrix-specific operations such as `dot`, `trace` etc. here, but currently in the core module they are also implemented in `core/ndarray` instead of `core/matrix`, so I think it's better to leave it as some future work, not to make this quite nice PR over-complex.

Reply:
agreed!

Reply:
Those functions are there because because Algodiff needs them. So make sure the Algodiff will be broken. I agree logically we should keep the functions in their own module rather than duplicating the signature, the reality is more complicated though.

I tried but failed honestly, but someone may come up smarter way to reorganise the function signature so both Algodiff is happy and Ndarray signature is clean. But for now, let's keep this PR compact and avoid making things too complicatged.